### DISCUSSION TITLE
Active fork being regularly updated and accepting PRs

### DISCUSSION THREAD

**Discussion by peterlewis** [wg-easy/wg-easy#496] [Show and tell]
Hi all! I am not meaning to undermine the sterling work that @WeeJeWel is doing, but I note that a number of PRs (including security updates) remain un-merged. I have forked and have merged in the following PRs:

https://github.com/WeeJeWel/wg-easy/pull/209
https://github.com/WeeJeWel/wg-easy/pull/319

I'd like to thank @pheiduck for his contributions, especially with regards to automatic builds and dependabot integration!

The repo can be found at https://github.com/peterlewis/wg-easy

Again, not meaning to undermine @WeeJeWel's work. Once the upstream repo sees more activity I can happily cease development on fork, and merge changes upstream, if desired.

**Comment by pheiduck** [comment]
> Again, not meaning to undermine @WeeJeWel's work. Once the upstream repo sees more activity I can happily cease development on fork, and merge changes upstream, if desired.

Good idea, I will also make a donation to @WeeJeWel  there and hope he resumes this project later.

**Comment by WeeJeWel** [comment]
Thanks :) I'm quite happy with the state of this project, and also very busy in my life, so it might take a while before PRs get merged.

**Comment by gaby** [comment]
@peterlewis Do you plan to merge new features into your fork? For example: dark mode, new env flags or is the plan to only keep dependencies/docker up-to-date?

**Reply by peterlewis** [reply]
Great question! I love the [recent PR for dark mode](https://github.com/WeeJeWel/wg-easy/pull/469) and would like to get this merged in as soon as I can. Keen to move the external tailwind dependency locally, but hadn't yet had the opportunity to do so. I've started looking at Tailwind, but been busy with other bits too.

I could ask @goodbyepavlyi if they don't mind doing so, but felt a bit cheeky as it's not my project!

Regarding the additional variables, the answer is yes, but there are a number of different PRs and have been keen to curate the most appropriate submissions. At the moment you could shift between @WeeJeWel's image and mine without having to make changes to underlying compose file, you see, which is good for compatibility.

Do you have any thoughts as to which are most appropriate to include or the direction that the fork should take?

**Reply by gaby** [reply]
I'm not sure which ENV's right now. Having dark mode would be a pretty cool feature. I do agree tailwindcss has to be running local. A lot of people use this project in networks without internet access.

Another feature that's lacking is support for DNS. This was partially covered by https://github.com/WeeJeWel/wg-easy/pull/323 but didnt get any traction after my comments last year.

**Reply by goodbyepavlyi** [reply]
Hello @peterlewis, I wanted to inform you of a recent commit where I have updated the Tailwind CSS implementation to be used locally instead of relying on a CDN.

**Reply by gaby** [reply]
@goodbyepavlyi awesome work!

**Reply by goodbyepavlyi** [reply]
@gaby thank you, I appreciate it.

**Reply by peterlewis** [reply]
That's fantastic! Thank you, @goodbyepavlyi! :) I'll have another look and merge in your PR, ASAP!

**Comment by thegabriele97** [comment]
Hi! I recently discovered this project and I am loving it. Wouldn't be better to accept maintainers to help you managing this awesome project?

**Reply by WeeJeWel** [reply]
People can fork it, and we'll see what gets more popular :)

For me, the current state works and I dislike projects that gets many features that aren't necessary (e.g. dark mode, options for the minority of users). But everyone is free to fork it.

**Comment by pheiduck** [comment]
Since @peterlewis fork has been inactive for 2 months. I continue to work on it, my fork will most likely contain maintenance, no major feature updates.

**Reply by gaby** [reply]
Since Sept 4 is not "inactive" 😂

**Reply by peterlewis** [reply]
Hullo! Sorry, not inactive, just distracted! Thank you for the gentle reminder, though! :-) Have merged the latest, I’m sorry that the changes thus far remain in nightly. The reason being is that, with the current version scheme, I was keen to ensure that the pending PRs were fully considered, before I bump the stable.

So long as you don’t mind risking it on nightly, I’ll resolve to be more present! :)

To add, I think one of the main reasons I have been slow to merge the new features, was considering what @WeeJeWel had said with regards to new features. I’m not at all averse, but want to be making sure that anything that is added is relevant to a wide audience. After all, thanks to his hard work, it is, at the moment, remarkably stable, and would be lovely to keep it that way! :-)

**Reply by peterlewis** [reply]
I will resolve, however, move the maintenance updates into stable this weekend and will aim to be more regular with keeping stable aligned with maintenance releases.

**Reply by gaby** [reply]
@peterlewis Thank you! 🥇

**Reply by pheiduck** [reply]
Okay, very nice than I will continue to make PRs for your repo ;)

**Reply by peterlewis** [reply]
They are very much appreciated! :-) Thank you!

**Reply by Hansaplast27** [reply]
A thx for your work first. I really enjoy using this.

I have moved over to peterlewis, as it seems the logical choice to stay updated. Although I must confess I find it hard to choose between pheiduck or peterlewis. Any insight might help others too.

I am just a user of the project, running it as a docker container within an ubuntu lxc on proxmox. Its an auto updating docker of the "latest". Just trying to make sure to stay on top of any security updates. I find that the most important part as wg-easy gets the only open port on my outward facing my network. 

Once again, thx for the work.

**Reply by peterlewis** [reply]
Thank you for your message @Hansaplast27! With regards to whether to choose peterlewis or pheiduck, I am keen to speak with @pheiduck to see if we can collaborate on a single fork. I sent him a message via Mastodon, but it may not have been received, or I may have got the wrong account...

**Reply by peterlewis** [reply]
Or, even better, whether @WeeJeWel would consider allowing us to both maintain this repo! :)

**Reply by WeeJeWel** [reply]
I'd be open for that @peterlewis! Have you seen my pinned discussion?

**Reply by pheiduck** [reply]
> Thank you for your message @Hansaplast27! With regards to whether to choose peterlewis or pheiduck, I am keen to speak with @pheiduck to see if we can collaborate on a single fork. I sent him a message via Mastodon, but it may not have been received, or I may have got the wrong account...

@peterlewis First of all, I apologize that you have not received an answer from me on Mastodon. I do not have Mastodon.
To your last answer here, I would very much appreciate it if we could continue to maintain the project together if @WeeJeWel accepts. :)

**Reply by peterlewis** [reply]
> I'd be open for that @peterlewis! Have you seen my pinned discussion?

Amazing news! Yes, I had seen discussion but had felt there may be others who were more even more well suited. There's nothing I'd love more than to put my fork into archive mode and to simply work with @pheiduck to maintain your upstream! 😄
